lib/src/platform/platform.dart

```diff
@@ -0,0 +1,16 @@
+export 'src/ably_event_message.dart';
```

Comment:
Originally, I would say there was still "implementation" inside the `spec` folder (the spec wasn't just abstract classes without implementation). I think the code in `impl` is specifically `platform` code, hence the rename. 

With the "mixing up", effectively, I moved all the `spec` files into the same level `impl` and renamed `impl` to `platform`. `platform` is still in its own folder/ module. So there is still a distinguishing feature, but less separate.

In federated plugins, the platform code would be in a separate package. 

Also, I think it would be easier to do file moves after the PN PR is merged (to avoid merge conflicts). We could consider moving all folder except `platform` into `non-platform`, because IMHO that is the only distinguishing feature between them. But maybe the folder names partly highlight this already?

Reply:
They do, but it is a little effortful to find out that `platform` is not a module similar to others (`push`, `rest`, `realtime`). 

I am unable to come up with any suggestion right now, but if needed, it's better to do after PN release.